CHANGELOG.md

```diff
@@ -10,6 +10,12 @@
 
 #### Enhancements
 
+* Introduce SwiftLintCommandPlugin.
+  Rename SwiftLintBuildToolPlugin.
```

Comment:
This is a breaking change and should therefore be mentioned in the `Breaking` section.

Reply:
Is it actually a breaking change if has not been released yet?

Reply:
The plugin has existed for about a year already.

Reply:
Oh, right 🤦 Sorry!

To avoid the breaking change then, we could consider the following:

- instead of `SwiftLintBuildToolPlugin`, use `SwiftLintPlugin` (as it has been named)
- instead of `SwiftLintCommandPlugin`, use `SwiftLintCommand` (without the word `plugin`)

@SimplyDanny Do you prefer `SwiftLintPlugin`/`SwiftLintCommand` (avoids breaking change, but not clear that the command is also a plugin) or `SwiftLintBuildToolPlugin`/`SwiftLintCommandPlugin` (breaking change, but makes it very clear what each one is)?

Reply:
As a third alternative, `SwiftLintPlugin` could be deprecated (printing a message) and   `SwiftLintBuildToolPlugin` could be introduced as the new name, having two `*Plugins` in the `Package.swift` for some time.

However, I actually like the `SwiftLintCommand` name. So of all those variants that's probably the easiest.

Reply:
Thinking more about this … the recent change to the build tool plugin could be considered a breaking change for [unexpected project structures](https://github.com/realm/SwiftLint#unexpected-project-structures) where the configuration file is located outside of the package/project directory.

Because of this, it could be seen as a benefit that a name change to `SwiftLintBuildToolPlugin` is also breaking, bringing attention to the plugin's revamped implementation.

I will leave it up to you both (@SimplyDanny and @garricn) to decide. I just wanted to surface this additional consideration so you were aware. Thank you 🙏

Reply:
The new names clearly distinguish that one is a build tool plugin and the other is a command plugin. It is unfortunate that it will require a change to - at least - the Package.swift files of consumers. However, I agree with @tinder-cfuller that this could be a benefit to make the plugin implementation change (which may be breaking) more apparent when consumers upgrade.

Do either of you anticipate some greater challenge posed by this breaking change than just simply updating the name in the Package.swift file? For example, it's also possible that `SwiftLintPlugin` is also referenced in the documentation of consumers. If so, then perhaps we should consider an alternative. If not, I don't see this breaking change posing a drastic challenge on upgrading consumers.

Nevertheless, I defer to the judgment of you all. Let me know your thoughts.

Reply:
I simply wanted to surface some options so an informed decision would be made. Thanks for considering the choices. Much appreciated. Thank you.

The new names are fine by me 👍

Reply:
The only problem is that people tend to ignore changelogs. But in this case, there is no other choice but to rename the plugin when updating the SwiftLint version as it just won't resolve. That's fine then. Let's stay with what you propose in the PR.

Reply:
Sounds good. Thank you for merging! I will be available for any Issues related to this command as well as the rename.